File: 10137.cpp
#include <iostream>
#include <cstdio>
#include <vector>
#include <cmath>
#include <algorithm>
using namespace std;

inline double truncate(double d)
{
  int tmp = d * pow(10, 2);
  return tmp / pow(10, 2);
}

inline double work(const vector<double> &v)
{
  double avg = 0.0;
  for (vector<double>::size_type i = 0;i != v.size();++i) {
    avg += v[i];
  }
  avg /= v.size();
  
  double pos = 0.0, neg = 0.0;
  for (vector<double>::size_type i = 0;i != v.size();++i) {
    double aux = v[i] - avg;
    (aux > 0) ? pos += truncate(aux): neg += truncate(-aux);
  }
  pos = truncate(pos);
  neg = truncate(neg);

  printf("~~~%.20f\n", pos);
  printf("~~~%.20f\n", neg);
  printf("~~~%.20f\n", fabs(pos - neg));
  //printf("~~~%.20f\n", truncate(0.01));
  //printf("~~~%.20f\n", fabs(fabs(pos - neg) - truncate(0.01)));
  if (truncate(0.01) < fabs(pos - neg)) {
  //if (fabs(fabs(pos - neg) - truncate(0.01)) > 0.001) {
    printf(">>>>><<<<<\n");
    return (pos + neg) * 0.5;
  } else {
    printf("<<<<<>>>>>\n");
    return truncate(min(pos, neg));
  }
}

int main()
{
  int n;
  vector<double> v;
  while(cin >> n && n) {
    double aux;
    v.reserve(n);
    for (int i = 0;i < n; i++){
      cin >> aux;
      v.push_back(aux);
    }
    printf("$%.2lf\n", work(v));
    v.clear();
  }
  return 0;
}

File: _10137.java
import java.util.Locale;
import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;
import java.math.BigDecimal;
import java.math.MathContext;
import java.math.RoundingMode;
import java.text.DecimalFormat;
import java.math.BigInteger;
import java.text.DecimalFormatSymbols;

public class _10137 {
    private static BigDecimal work(List<BigDecimal> v) {
        BigDecimal avg = new BigDecimal(0);
        for (BigDecimal i: v) {
            avg = avg.add(i);
        }
        avg = avg.divide(new BigDecimal(v.size()), RoundingMode.HALF_DOWN);
        System.out.println(">>> " + avg);

        BigDecimal pos = new BigDecimal(0),
            neg = new BigDecimal(0); 
        for (BigDecimal i: v) {
            BigDecimal aux = i.subtract(avg);
            if (aux.compareTo(new BigDecimal(0)) == 1) {
                pos = pos.add(aux);
            } else {
                neg = neg.subtract(aux);
            }
        }

        if (pos.subtract(neg).abs().compareTo(new BigDecimal("0.1")) == 1) {
            return pos.add(neg).divide(new BigDecimal(2), RoundingMode.DOWN).setScale(2, RoundingMode.DOWN);
        } else {
            return pos.min(neg).setScale(2, RoundingMode.DOWN);
            }
    }

    public static void main(String args[]) {
        int n;
        List<BigDecimal> v 
            = new ArrayList<BigDecimal>();
        Scanner in = new Scanner(System.in);
        DecimalFormat formatter = (DecimalFormat)DecimalFormat.getInstance(new Locale("en_US"));
        DecimalFormatSymbols symbols = formatter.getDecimalFormatSymbols();
        symbols.setGroupingSeparator('#');
        formatter.setDecimalFormatSymbols(symbols);
        in.useLocale(new Locale("C"));
        n = in.nextInt();
        while(n != 0) {
            for (int i = 0; i < n; i++){
                v.add(in.nextBigDecimal());
            }
            BigDecimal bd = work(v);
            long remainder = bd.multiply(new BigDecimal(100)).longValue() % 100;
            System.out.println("$" + formatter.format(bd.longValue()).replace("#", "") + "." + String.format("%02d", new Integer((int)remainder)));
            v.clear();
            n = in.nextInt();
        }
    }

    private static MathContext mathContext
        = new MathContext(3, RoundingMode.DOWN);
}
